### PR TITLE
[Mime] Add a way to control the HTML to text conversion

### LINKED ISSUE #46295: Make `Mime\BodyRenderer` markdown conversion configurable
### Description

Symfony can already automatically generate a text version of an e-mail in `\Symfony\Bridge\Twig\Mime\BodyRenderer` using [league/html-to-markdown](https://github.com/thephpleague/html-to-markdown) (see https://symfony.com/doc/current/mailer.html#text-content)

But there is no way to configure the conversion options (e.g. we would like to remove the `img` tags as well). The instantiation is hardcoded in the constructor and the `BodyRenderer` itself is marked as final:
https://github.com/symfony/symfony/blob/1ac98fc9c619a1202e38018e7c578869ad7174d0/src/Symfony/Bridge/Twig/Mime/BodyRenderer.php#L23-L37
(The only way I see as an extension option is to copy the `BodyRenderer` to `MyBodyRenderer`, modify it and try to keep up with the changes in Symfony)

I suggest that the `League\HTMLToMarkdown\HtmlConverter` is wrapped behind some `HtmlToTextConverter` interface, so it would be possible to either replace it with other convertor or at least configure it differently.

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| Tickets       | Fix #46295 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bridge/Twig/composer.json [hunk 1]
```diff
@@ -59,6 +60,7 @@
         "symfony/form": "<6.1",
         "symfony/http-foundation": "<5.4",
         "symfony/http-kernel": "<6.2",
+        "symfony/mime": "<6.2",
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/composer.json:63]: should be updated in require-dev also

File: src/Symfony/Bridge/Twig/Mime/BodyRenderer.php [hunk 2]
```diff
@@ -24,19 +27,13 @@ final class BodyRenderer implements BodyRendererInterface
...
         $this->twig = $twig;
         $this->context = $context;
-        if (class_exists(HtmlConverter::class)) {
-            $this->converter = new HtmlConverter([
-                'hard_break' => true,
-                'strip_tags' => true,
-                'remove_nodes' => 'head style',
-            ]);
-        }
+        $this->converter = $converter ?: (interface_exists(HtmlConverterInterface::class) ? new LeagueHtmlToMarkdownConverter() : new DefaultHtmlToTextConverter());
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Mime/BodyRenderer.php:36]: `interface_exists(HtmlConverterInterface::class)` really? Should this check for a League* class?

File: src/Symfony/Component/Mime/HtmlToTextConverter/LeagueHtmlToMarkdownConverter.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+use League\HTMLToMarkdown\HtmlConverter;
+use League\HTMLToMarkdown\HtmlConverterInterface;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class LeagueHtmlToMarkdownConverter implements HtmlToTextConverterInterface
+{
+    public function __construct(
+        private ?HtmlConverterInterface $converter = null,
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Mime/HtmlToTextConverter/LeagueHtmlToMarkdownConverter.php]:
```
private HtmlConverterInterface $converter = new HtmlConverter([
    'hard_break' => true,
    'strip_tags' => true,
    'remove_nodes' => 'head style',
]),
```

File: src/Symfony/Component/Mime/HtmlToTextConverter/HtmlToTextConverterInterface.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+interface HtmlToTextConverterInterface
+{
+    /**
+     * Converts and HTML representation of a Message to a text representation.
+     */
+    public function convert(string $html): string;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Mime/HtmlToTextConverter/HtmlToTextConverterInterface.php]: do we need to specify something about the charset, eg "the output must be UTF-8"?

File: src/Symfony/Bridge/Twig/Mime/BodyRenderer.php (hunk 2 above)
Comment (fabpot) [reply src/Symfony/Bridge/Twig/Mime/BodyRenderer.php:36]: Yes, see `use League\HTMLToMarkdown\HtmlConverterInterface`, that interface IS the one from League ;)

File: src/Symfony/Component/Mime/HtmlToTextConverter/HtmlToTextConverterInterface.php (hunk 4 above)
Comment (fabpot) [reply src/Symfony/Component/Mime/HtmlToTextConverter/HtmlToTextConverterInterface.php]: See the second commit, I've done something a bit different.

**Review by nicolas-grekas — APPROVED** [review]

**Pull request merged**